tests/test_geoprocessing.py

```diff
@@ -1011,38 +1011,22 @@ def test_zonal_stats_for_small_polygons(self):
 
     def test_zonal_stats_no_bb_overlap(self):
         """PGP.geoprocessing: test no vector bb raster overlap."""
-        gpkg_driver = ogr.GetDriverByName('GPKG')
         vector_path = os.path.join(self.workspace_dir, 'vector.gpkg')
-        vector = gpkg_driver.CreateDataSource(vector_path)
-
-        srs = osr.SpatialReference()
-        srs.ImportFromEPSG(4326)
-        layer = vector.CreateLayer('small_vector', srs=srs)
-        layer_defn = layer.GetLayerDefn()
-
         # make an n x n raster with 2*m x 2*m polygons inside.
         pixel_size = 1
         subpixel_size = pixel_size / 5
         origin_x = 1
         origin_y = -1
         n = 16
-        layer.StartTransaction()
         x_pos = origin_x - n
         y_pos = origin_y + n
-        shapely_feature = shapely.geometry.Polygon([
+        geom = shapely.geometry.Polygon([
             (x_pos, y_pos),
             (x_pos+subpixel_size, y_pos),
             (x_pos+subpixel_size, y_pos-subpixel_size),
             (x_pos, y_pos-subpixel_size),
             (x_pos, y_pos)])
-        new_feature = ogr.Feature(layer_defn)
-        new_geometry = ogr.CreateGeometryFromWkb(shapely_feature.wkb)
-        new_feature.SetGeometry(new_geometry)
-        layer.CreateFeature(new_feature)
-        layer.CommitTransaction()
-        layer.SyncToDisk()
-        layer = None
-        vector = None
+        _geometry_to_vector([geom], vector_path, projection_epsg=4326)

```

Comment:
Updating a few unrelated tests to use `_geometry_to_vector` - better to avoid duplicating all the layer creation code, and it sets the geometry type correctly, which is now required.